File: implementations/java/Generate.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Generate {
    public static void main(String[] args) throws IOException {
        Files.createDirectories(CidUtil.CIDS_DIR);
        for (Path path : Files.list(CidUtil.EXAMPLES_DIR).toList()) {
            if (Files.isDirectory(path)) {
                continue;
            }
            byte[] content = CidUtil.readFile(path);
            String cid = CidUtil.computeCid(content);
            Path destination = CidUtil.CIDS_DIR.resolve(cid);
            CidUtil.writeFile(destination, content);
            System.out.printf("Wrote %s from %s%n", cid, path.getFileName());
        }
    }
}

File: implementations/haskell/generate.hs
module Main where

import CID (cidsDir, computeCID, examplesDir)
import Control.Monad (forM_, when)
import Data.List (sort)
import qualified Data.ByteString as BS
import System.Directory (createDirectoryIfMissing, doesFileExist, listDirectory)
import System.FilePath ((</>))

main :: IO ()
main = do
  createDirectoryIfMissing True cidsDir
  entries <- sort <$> listDirectory examplesDir
  forM_ entries $ \name -> do
    let path = examplesDir </> name
    isFile <- doesFileExist path
    when isFile $ do
      content <- BS.readFile path
      let cid = computeCID content
          destination = cidsDir </> cid
      BS.writeFile destination content
      putStrLn $ "Wrote " ++ cid ++ " from " ++ name
